awx/main/utils/external_logging.py

```diff
@@ -57,7 +57,11 @@ def construct_rsyslog_conf_template(settings=settings):
             'action.resumeRetryCount="-1"',
             'template="awx"',
             'errorfile="/var/log/tower/rsyslog.err"',
-            f'action.resumeInterval="{timeout}"'
+            f'action.resumeInterval="{timeout}"',
+            'queue.filename="awx-external-log"',
```

Comment:
This causes rsyslog to use a disk-assisted memory queue:

https://rsyslog-doc.readthedocs.io/en/latest/concepts/queues.html#disk-assisted-memory-queues

> If a disk queue name is defined for in-memory queues (via $<object>QueueFileName), they automatically become “disk-assisted” (DA). In that mode, data is written to disk (and read back) on an as-needed basis.

> Most importantly, the disk queue is activated if the primary queue is full or needs to be persisted on shutdown. Disk-assisted queues combine the advantages of pure memory queues with those of  pure disk queues. Under normal operations, they are very fast and messages will never touch the disk. But if there is need to, an unlimited amount of messages can be buffered (actually limited by free disk space only) and data can be persisted between rsyslogd runs.

Reply:
tl;dr for this PR:

Hold onto at most 100k events, or 1G of event data (whichever comes first).

The goal of this isn't to 100% guarantee *all* message delivery, but to give users a reasonable buffer if e.g., their Splunk has a network blip, or even if it goes down for an hour.